app/controllers/foreman_inventory_upload/uploads_controller.rb

```diff
@@ -11,14 +11,18 @@ def last
 
     def download_file
       filename = ForemanInventoryUpload.facts_archive_name(params[:organization_id])
-      path = Rails.root.join(ForemanInventoryUpload.uploads_folder, filename)
-      unless File.exist? path
-        return throw_flash_error(
-          "Path doesn't exist: #{path}"
-        )
-      end
+      uploads_file_path = ForemanInventoryUpload.uploads_file_path(filename)
+      done_file_path = ForemanInventoryUpload.done_file_path(filename)
 
-      send_file path, disposition: 'attachment', filename: filename
+      unless File.exist? uploads_file_path
+        unless File.exist? done_file_path
+          return throw_flash_error(
+            "File doesn't exist in: #{uploads_file_path} -or- #{done_file_path}"
+          )
+        end
+        return send_file done_file_path, disposition: 'attachment', filename: filename
+      end
+      send_file uploads_file_path, disposition: 'attachment', filename: filename
```

Comment:
How about putting the paths into an array, trying to `#find` first one which exists on the filesystem and if there is none found, returning the flash error with message generated from the array?

Reply:
While we are at it, we can use `Dir.glob` do search for a file.

Reply:
what do you think about this?
@ShimShtein where should we use the `Dir.glob`? Thanks

```rb
def download_file
      filename = ForemanInventoryUpload.facts_archive_name(params[:organization_id])
      file_paths = [
        ForemanInventoryUpload.uploads_file_path(filename),
        ForemanInventoryUpload.done_file_path(filename)
      ]
 
      final_path = file_paths.find do |path|
        path unless path.empty?
      end
      return send final_path, disposition: 'attachment', filename: filename} unless final_path.nil?
      throw_flash_error(
        "File doesn't exist in: #{file_paths.join(',')}"
      )
    end
```

Reply:
Instead of the array and find, you could do something like `Dir["#{ForemanInventoryUpload.uploads_folder}/**/#{filename}]` and take a first from it. Given how things currently work, the `Dir[...]` should give you back an array with 0 or 1 elements.

Reply:
@laviro we can have the `final_path` in one line, since the block contains only one line something like: `final_path = file_paths.find { |path| path unless path.empty? }`

What do you think?

I have not tested this code but this code will only work if `ForemanInventoryUpload.uploads_file_path(filename) and ForemanInventoryUpload.done_file_path(filename)` gives a empty string (in the case of a failure), but if it gives nil as a value then this code might fail.

Reply:
Thanks everyone, changed into this:
```rb
def download_file
  filename = ForemanInventoryUpload.facts_archive_name(params[:organization_id])
  files = Dir["#{ForemanInventoryUpload.uploads_folder}**/#{filename}"]

  return send_file files[0], disposition: 'attachment', filename: filename unless files.empty?
  throw_flash_error "File doesn't exist"
end
```